src/parser.ts

```diff
@@ -306,17 +308,59 @@ function parseSuperTypes(
   options: Options,
   processed: Processed,
   usedNames: UsedNames
-): TNamedInterface[] {
+): TSuperType[] {
   // Type assertion needed because of dereferencing step
   // TODO: Type it upstream
   const superTypes = schema.extends as SchemaSchema | SchemaSchema[] | undefined
   if (!superTypes) {
     return []
   }
   if (Array.isArray(superTypes)) {
-    return superTypes.map(_ => newNamedInterface(_, options, _, processed, usedNames))
+    return superTypes.map(_ => getOrCreateSuperType(_, options, _, processed, usedNames))
+  }
+  return [getOrCreateSuperType(superTypes, options, superTypes, processed, usedNames)]
+}
+
+function isSuperType(type: any): type is TSuperType {
+  return type.type === 'CUSTOM_TYPE' || type.type === 'INTERFACE'
+}
+
+function getOrCreateSuperType(
+  schema: SchemaSchema,
+  options: Options,
+  rootSchema: JSONSchema,
+  processed: Processed,
+  usedNames: UsedNames
+): TSuperType {
+  if (processed.has(schema)) {
```

Comment:
What sort of cycle is this catching? Mind adding a test case for it?

Reply:
This just protects against programming mistakes. At this point schema should've been processed already. If it hasn't, something went wrong earlier. Guess I could throw an error in that case

Reply:
Added the error

Reply:
The function is a get or create. If processed has schema, it's a get, otherwise it's a create. The tests already cover both branches